Library/MapleBacon/MapleBacon/Storage/DiskStorage.swift

```diff
@@ -6,25 +6,29 @@ import UIKit
 
 public final class DiskStorage {
 
+    /// Singleton instance
     public static let sharedStorage = DiskStorage()
-
-    private static let QueueLabel = "de.zalando.MapleBacon.Storage"
-
-    private let fileManager = NSFileManager.defaultManager()
-    private let storageQueue = dispatch_queue_create(QueueLabel, DISPATCH_QUEUE_SERIAL)
+    private let useUuid: Bool = MapleBaconConfig.sharedConfig.storage.useUUID
+    
+    /// label for serial queue
+    private static let queueLabel = MapleBaconConfig.sharedConfig.storage.queueLabel
+    private let fileManager = FileManager.default()
+    private let storageQueue = DispatchQueue(label: queueLabel, attributes: .serial, target: nil)
     private let storagePath: String
-    public var maxAge: NSTimeInterval = 60 * 60 * 24 * 7
-
+    public var maxAge: TimeInterval = 60 * 60 * 24 * 7
 
     public convenience init() {
-        self.init(name: "default")
+        self.init(name: MapleBaconConfig.sharedConfig.storage.defaultStorageName)
```

Comment:
The way this config singleton is used it's possible to mutate it while the storage is using it. I think a factory would be better suited here where you can it to construct a default or custom configuration for you that's passed to the storage.